builder/src/prepare_project.py

```diff
@@ -151,7 +291,263 @@ def supports(self, version):
                 StrictVersion(self.version) >= StrictVersion(version))
 
 
+class PublishedApp(object):
+    """Represents a published app.
+
+    This class represetns an app that has already been published. This
+    is the simplest clase for an app. A very simple Dockerfile can be
+    generated.
+    """
+
+    # Dockerfile template to be used when packaging up published apps.
+    DOCKERFILE_CONTENTS = textwrap.dedent(
+        """\
+        FROM {runtime_image}
+        ADD ./ /app
+        ENV ASPNETCORE_URLS=http://*:${{PORT}}
+        WORKDIR /app
+        ENTRYPOINT [ "dotnet", "{dll_name}.dll" ]
+        """)
+
+    def __init__(self, root):
+        """Initializes the PublishedApp instance.
+
+        Args:
+            root: A string with the path to the root of the published
+                  app.
+        """
+        self.root = root
+        self.deps_path = get_deps_path(root)
+
+    def generate_dockerfile(self, version_map, output):
+        """Generates the Dockerfile for this app.
+
+        Generates a simple Dockerfile that will wrap an already
+        published app. This Dockerfile will just take the files as is,
+        no build will be attempted.
+
+        Args:
+            version_map: A dictionary that maps versions of the
+                         runtime to the base Docker image to use.
+            output: A string with the path where to save the resulting
+                    Dockerfile.
+        """
+        minor_version = self._get_runtime_minor_version()
+        if minor_version is None:
+            print('No valid .NET Core runtime version found for the app or it is not a ' +
+                  'supported app.')
+            sys.exit(1)
+
+        base_image = get_base_image(version_map, minor_version)
+        if base_image is None:
+            print('The app requires .NET Core runtime version {0} which is not supported at ' +
+                  'this time.').format(minor_version)
+            sys.exit(1)
+
+        project_name = self._get_project_assembly_name()
+        assembly_name = ASSEMBLY_NAME_TEMPLATE.format(project_name)
+        if not os.path.isfile(os.path.join(self.root, assembly_name)):
+            print('Cannot find entry point assembly {0} for ASP.NET Core project'
+                  .format(assembly_name))
+            sys.exit(1)
+
+        contents = PublishedApp.DOCKERFILE_CONTENTS.format(runtime_image=base_image.image,
+                                                           dll_name=project_name)
+        with open(output, 'wt') as out:
+            out.write(contents)
+
+    def _get_project_assembly_name(self):
+        """Returns the name of the entrypoint assembly given the .deps.json
+        file name.
+
+        Returns:
+            The name of the entry point assembly.
+        """
+        filename = os.path.basename(self.deps_path)
+        return filename[:-len(DEPS_EXTENSION)]
```

Comment:
This won't work because the extension is a composite ".deps.json" and I am looking for the text before that. So in the file `MyProject.deps.json` I'm looking for `MyProject`.